contracts/fee/FeeHandler.sol

```diff
@@ -3,46 +3,208 @@
+        }
+
+        uint256 batchSize = dataStore.getUint(Keys.buybackBatchAmountKey(buybackToken));
+        _validateBuybackToken(batchSize, buybackToken);
+
+        uint256 availableFeeAmount = _getAvailableFeeAmount(feeToken, buybackToken);
+        uint256 maxFeeTokenAmount = _getMaxFeeTokenAmount(feeToken, buybackToken, batchSize);
+        if (availableFeeAmount >= maxFeeTokenAmount) {
+            _buybackFees(feeToken, buybackToken, batchSize, maxFeeTokenAmount, availableFeeAmount);
+        } else if (availableFeeAmount >= minOutputAmount) {
+            _buybackFees(feeToken, buybackToken, batchSize, availableFeeAmount, availableFeeAmount);
+        } else {
+            revert Errors.InsufficientBuybackOutputAmount(feeToken, buybackToken, availableFeeAmount);
+        }
+    }
+
+    function getOutputAmount(
+        address[] calldata markets,
+        address feeToken,
+        address buybackToken
+    ) external view returns (uint256) {
+        uint256 batchSize = dataStore.getUint(Keys.buybackBatchAmountKey(buybackToken));
+        _validateBuybackToken(batchSize, buybackToken);
 
+        uint256 feeAmount;
+        FeeAmounts memory feeAmounts;
+        uint256 availableFeeAmount = _getAvailableFeeAmount(feeToken, buybackToken);
         for (uint256 i; i < markets.length; i++) {
-            FeeUtils.claimFees(
-                dataStore,
-                eventEmitter,
-                markets[i],
-                tokens[i],
-                receiver
-            );
+            if (markets[i] == address(0)) {
+                feeAmount = IVaultV1(vaultV1).feeReserves(feeToken);
+                feeAmounts = _getFeeAmounts(1, feeAmount);
+            } else {
+                feeAmount = dataStore.getUint(Keys.claimableFeeAmountKey(markets[i], feeToken));
+                feeAmounts = _getFeeAmounts(2, feeAmount);
+            }
+
+            feeAmount = buybackToken == gmx ? feeAmounts.gmx : feeAmounts.wnt;
+            availableFeeAmount = availableFeeAmount + feeAmount;
+        }
+
+        uint256 maxFeeTokenAmount = _getMaxFeeTokenAmount(feeToken, buybackToken, batchSize);
+        if (availableFeeAmount >= maxFeeTokenAmount) {
+            return maxFeeTokenAmount;
+        } else {
+            return availableFeeAmount;
+        }
+    }
+
+    function _buybackFees(
+        address feeToken,
+        address buybackToken,
+        uint256 batchSize,
+        uint256 buybackAmount,
+        uint256 availableFeeAmount
+    ) private {
+        IERC20(feeToken).safeTransfer(msg.sender, buybackAmount);
+        IERC20(buybackToken).safeTransferFrom(msg.sender, address(this), batchSize);
+
+        availableFeeAmount = availableFeeAmount - buybackAmount;
+        _setAvailableFeeAmount(feeToken, buybackToken, availableFeeAmount);
+    }
+
+    function _setAvailableFeeAmounts(uint256 version, address feeToken, uint256 feeAmount) private {
```

Comment:
would be clearer to name this _incrementAvailableFeeAmounts to differentiate it from _setAvailableFeeAmount

Reply:
No problem, I’ve renamed the function for more clarity.